### PR TITLE
Added Required to the Schema for rendering.

### PR DESCRIPTION
**Pull request by robert-sykes**
The `Required` option does not seem to be rendering for a property. I am not sure if this is intentional or not. This PR adds the support for it to be rendered out with `required` being defaulted to true. This allows `required` properties to be marked as such.

### CODE REVIEW

**Comment by Ferror** [comment]
I'm going to reopen the PR because of CI

**Comment by Ferror** [comment]
There is an issue in asyncapi CLI tool that does not exit code 1 when schema is invalid - https://github.com/asyncapi/cli/pull/1068

Yet, if you look into the pipeline https://github.com/Ferror/asyncapi-doc-bundle/actions/runs/7544841545/job/20539100621?pr=3 you will notice errors. Mostly because the `required` parameters in schema must not be produced as a property `required: true`, but as an array of properties. Example:

```yaml
asyncapi: 2.6.0
info:
  title: 'Service Example API'
  version: 1.2.3
  description: 'This service is in charge of processing user signups'
servers:
  production:
    url: broker.mycompany.com
    protocol: amqp
    description: 'This is production broker.'
  staging:
    url: broker.mycompany.com
    protocol: amqp
    description: 'This is staging broker.'
channels:
  product.created:
    subscribe:
      message:
        $ref: '#/components/messages/ProductCreated'
components:
  messages:
    ProductCreated:
      payload:
        type: object
        properties:
          id:
            type: integer
            description: ''
          amount:
            type: number
            description: ''
          currency:
            type: string
            description: ''
          isPaid:
            type: boolean
            description: ''
          createdAt:
            type: string
            description: ''
            format: date-time
          week:
            type: integer
            description: ''
          payment:
            type: string
            description: ''
          products:
            type: string
            description: ''
          tags:
            type: string
            description: ''
        required:
          - id
          - currency
          - isPaid

```

**Comment by Ferror** [comment]
So you are right about the issue, but the solution still needs to be there. Let me know if you would like to continue on this PR or I can provide fix for it 😄

File: src/Schema.php [hunk 1]
```diff
@@ -24,6 +24,10 @@ public function render(array $document): array
             if (isset($property['example'])) {
                 $properties[$property['name']]['example'] = $property['example'];
             }
+
```
Comment (Ferror) [inline src/Schema.php:28]: check out this:

```php
    public function render(array $document): array
    {
        $properties = [];
        $required = [];

        foreach ($document['properties'] as $property) {
            $properties[$property['name']]['type'] = PropertyTypeTranslator::translate($property['type']);

            if (isset($property['description'])) {
                $properties[$property['name']]['description'] = $property['description'];
            }

            if (isset($property['format'])) {
                $properties[$property['name']]['format'] = $property['format'];
            }

            if (isset($property['example'])) {
                $properties[$property['name']]['example'] = $property['example'];
            }

            if (isset($property['required']) && $property['required']) {
                $required = $property['name'];
            }
        }

        $message[$document['name']] = [
            'payload' => [
                'type' => 'object',
                'properties' => $properties,
                'required' => $required,
            ],
        ];

        return $message;
    }
```

**Comment by robert-sykes** [comment]
> So you are right about the issue, but the solution still needs to be there. Let me know if you would like to continue on this PR or I can provide fix for it 😄 

Yep. My mistake. I actually fixed this on my local generator or EventCatalog but completely forgot to come back to fix it here. 😂 happy to apply the fix if you haven't already. 

Had myself convinced it was required on the property and not an array of required properties. 🤦

**Comment by Ferror** [comment]
@robert-sykes go ahead. FYI. I'm starting slowly to build similar library for Java/Spring if you are interested 😄

File: src/Schema.php (hunk 1 above)
Comment (robert-sykes) [reply src/Schema.php:28]: I think this should fix it to work as expected.

**Comment by robert-sykes** [comment]
> @robert-sykes go ahead. FYI. I'm starting slowly to build similar library for Java/Spring if you are interested 😄

Ah - i'm afraid I don't know much about Java/Spring to be of any help. 😄

**Comment by robert-sykes** [comment]
@Ferror I think the following line: 
`return json_encode($this->generator->generate(), JSON_THROW_ON_ERROR | JSON_FORCE_OBJECT);` 
is causing the items in the array to render as:

```javascript
  "required": {
        "0": "name",
        "1": "email",
        "2": "age",
        "3": "isCitizen"
    }
```
and is causing the validation error we are seeing in the actions. I am not sure how get around this at the moment, only once test seems to fail if we remove the `JSON_FORCE_OBJECT` but I don't know what knock on effects this might have on the rendering. 

I've made the change and pushed the fix anyway to check that the Actions are working for now.

**Comment by Ferror** [comment]
@robert-sykes I will accept the PR and create issue to look into JSON Generator. It is durable, but I think we will need to hack a little bit ;p

**Review by Ferror — APPROVED** [review]

**Pull request merged**